tests/unit_tests/RocketStateTest.h

```diff
@@ -0,0 +1,262 @@
+//
+// Created by caleb on 29/11/22.
+//
+
+#ifndef JSB_STATE_ROCKETSTATETEST_H
+#define JSB_STATE_ROCKETSTATETEST_H
+
+#include <cxxtest/TestSuite.h>
+#include <FGFDMExec.h>
+#include "initialization/FGInitialCondition.h"
+
+using namespace JSBSim;
+
+
+class RocketStateTest : public CxxTest::TestSuite{
+private:
+    FGFDMExec setupTest() {
+        cout<<"\n-----------------------------"<<endl;
+        std::string aircraft_path_ = "/home/caleb/repos/jsbsim-state/externals/vehicle-models";
+        std::string model_name_ = "damk2-rocket";
+        std::string scene_path_ = "/home/caleb/repos/jsbsim-state/externals/vehicle-models/damk2-rocket/NZGT_ALT.xml";
```

Comment:
The test scripts will need to be modified a bit to use a somewhat generic vehicle + engine combination that is alright to add to a public repository, otherwise these tests will always fail.